torch/distributed/tensor/parallel/fsdp.py

```diff
@@ -331,7 +331,7 @@ def pre_load_state_dict_transform(
 
     _set_fsdp_extensions(DTensorExtensions())
 
-    def is_available() -> bool:
+    def enable_2d_w_fsdp() -> bool:
```

Comment:
Actually, I think we should move the whole try catch block inside this function directly (i.e. the whole extension registration should happen in this API directly)? 

Also we should add docstr to this function as it's public API.